libraries-llms/src/test/java/com/baeldung/langchain/ChainWithDocumentLiveTest.java

```diff
@@ -0,0 +1,67 @@
+package com.baeldung.langchain;
+
+import static dev.langchain4j.data.document.FileSystemDocumentLoader.loadDocument;
+import static java.time.Duration.ofSeconds;
+import static org.junit.Assert.assertNotNull;
+
+import java.nio.file.Paths;
+
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import dev.langchain4j.chain.ConversationalRetrievalChain;
+import dev.langchain4j.data.document.Document;
+import dev.langchain4j.data.document.splitter.DocumentSplitters;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.memory.chat.MessageWindowChatMemory;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.embedding.AllMiniLmL6V2EmbeddingModel;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.model.input.PromptTemplate;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+import dev.langchain4j.retriever.EmbeddingStoreRetriever;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.store.embedding.EmbeddingStoreIngestor;
+import dev.langchain4j.store.embedding.inmemory.InMemoryEmbeddingStore;
+
+public class ChainWithDocumentLiveTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(ChainWithDocumentLiveTest.class);
+
+    @Test
+    public void givenChainWithDocument_whenPrompted_thenValidResponse() {
+
```

Comment:
As per guidelines, avoid blank lines after opening braces and before ending braces. Apply this to all your sample code.

Reply:
Fixed.